jsapp/js/components/activity/activityLogs.query.ts

```diff
@@ -1,116 +1,85 @@
-
-  const metadata: ActivityLogsItem['metadata'] = {
-    source,
-    asset_uid,
-    ip_address,
-    log_subtype: log_subtype as AuditSubTypes,
-  };
-
-  if (action === 'update-name') {
-    metadata.old_name = 'I kwno somethign';
-    metadata.new_name = 'I know something';
-  }
-  if (action === 'deploy' || action === 'redeploy') {
-    metadata.latest_deployed_version_id = 'asd123f3fz';
-  }
-  if (action === 'replace-form' || action === 'update-form') {
-    metadata.latest_version_id = 'aet4b1213c';
-  }
-  if (
-    action === 'add-user' ||
-    action === 'remove-user' ||
-    action === 'update-permission' ||
-    action === 'transfer'
-  ) {
-    metadata.second_user = 'Josh';
-  }
-
-  return {
-    user,
-    user_uid,
-    username,
-    action: action as AuditActions,
-    log_type: log_type as AuditTypes,
-    metadata: metadata,
-  };
-};
-
-const curDate = new Date();
-const mockData: ActivityLogsItem[] = Array.from({length: 150}, (_, index) => {
-  curDate.setTime(curDate.getTime() - Math.random() * 1000000);
-  return {
-    id: index,
-    ...getRandomMockDescriptionData(),
-    date_created: moment(curDate).format('YYYY-MM-DD HH:mm:ss'),
-  };
-});
-// END OF MOCK GENERATION
-// =============================================================================
+import {fetchGet} from 'jsapp/js/api';
+import {endpoints} from 'jsapp/js/api.endpoints';
+import type {PaginatedQueryHookParams} from 'jsapp/js/universalTable/paginatedQueryUniversalTable.component';
 
 /**
  * Fetches the activity logs from the server.
  * @param limit Pagination parameter: number of items per page
  * @param offset Pagination parameter: offset of the page
  */
-const getActivityLogs = async (limit: number, offset: number) =>
-  new Promise<PaginatedResponse<ActivityLogsItem>>((resolve) => {
-    setTimeout(
-      () =>
-        resolve({
-          next: null,
-          previous: null,
-          count: mockData.length,
-          results: mockData.slice(offset, offset + limit),
-        } as PaginatedResponse<ActivityLogsItem>),
-      1000
-    );
+const getActivityLogs = async ({
+  projectId,
+  actionFilter,
+  limit,
+  offset,
+}: {
+  projectId: string;
+  actionFilter: string;
+  limit: number;
+  offset: number;
+}) => {
+  const params = new URLSearchParams({
+    limit: limit.toString(),
+    offset: offset.toString(),
   });
+  if (actionFilter) {
+    params.append('q', `action:${actionFilter}`);
+  }
+
+  const endpointUrl = endpoints.ASSET_HISTORY.replace(':asset_id', projectId);
+
+  return await fetchGet<PaginatedResponse<ActivityLogsItem>>(
+    `${endpointUrl}?${params}`,
+    {
+      errorMessageDisplay: t('There was an error getting one-time add-ons.'),
+    }
+  );
+};
 
 /**
  * Fetches the filter options for the activity logs.
+ *
+ * Filter options, for now, comes from ../api/v2/assets/[ assetId]]/history
```

Comment:
```suggestion
 * Filter options, for now, comes from `endpoints.ASSET_HISTORY`
```
I think this would be good enough. And if you want to keep the url string here, it should be exactly the same as the one from endpoints file

Reply:
I was trying to mean that it's documented in the API... maybe I need to rewrite better! 🤔